clocksilencer.ino

```diff
@@ -1,100 +1,111 @@
-    mp3.volume(5);  //Set volume value. From 0 to 30
-    delay(1000);
-    mp3.play(3); //this does not work. dunno why
-    Serial.println("Play OK signal");
-    //this also shows that no music is played.
-    
-  } else {
-    setenv("TZ", "CET-1CEST,M3.5.0,M10.5.0/3", 3);  //repair timezone
+  Serial.println("WiFi connected");
+  //---------set with NTP---------------
+  configTime(3600, 3600, "europe.pool.ntp.org");
+  struct tm timeinfo;
+  if (getLocalTime(&timeinfo)) {
+    rtc.setTimeStruct(timeinfo);
+  }
+  Serial.println("Attempt to update RTC");
+  while (rtc.getYear() == 1970) {
+    Serial.println("RTC updating in progress");
+  } 
+ WiFi. disconnect();
+ WiFi.mode(WIFI_OFF); //disconnect after synchronisation to save power
+ rtcTimeWasAlreadySetFromNTP=true; //time was updated
+ Serial.println ("WiFi turned off");
+ }else{ setenv("TZ", "CET-1CEST,M3.5.0,M10.5.0/3", 3);  //repair timezone
     tzset();
   }
-  
 }
-
-
 void loop() {
-
-  Serial.println(rtc.getTimeDate(true));  //  (String) 15:24:38 Sunday, January 17 2021
+  Serial.println(
+    rtc.getTimeDate(true)); //  (String) 15:24:38 Sunday, January 17 2021
 
   myDow = rtc.getDayofWeek();
   mySec = rtc.getSecond();
   myMin = rtc.getMinute();
   myHour = rtc.getHour(true);
 
-  if (mySec == 0 and myMin == 0) {     //at full hour
-    if (myHour > 7 and myHour < 19) {  //between 9 and 18
+ if ((59-myMin >5) or (29-myMin >5)){
+ //Here count the timeof sleep 
+   deepsleep();
+ }
+
+  if (mySec == 0 and myMin == 0) { //at full hour
+    if (myHour > 7 and myHour < 19) { //between 9 and 18
       //here another condition
       ringer(myHour);
     }
   }
-  if (mySec == 0 and myMin == 30) {    //at half hour
-    if (myHour > 7 and myHour < 19) {  //between 9 and 18
+  if (mySec == 0 and myMin == 30) { //at half hour
+    if (myHour > 7 and myHour < 19) { //between 9 and 18
       //here another condition
       ringer(1);
     }
   }
+
   struct tm timeinfo = rtc.getTimeStruct();
-  esp_sleep_enable_timer_wakeup(750000);  //in microseconds, but not the full second (1000000)
-  Serial.println("Deepsleep");
-  esp_deep_sleep_start();
+  
+  delay(1000);
 }
 void ringer(int myhours2) {
+  //run mosfet here
+  digitalWrite(12,1); //mosfet
+  hwSerial.begin(9600, SERIAL_8N1, 16, 17); 
+  myDFPlayer.begin(hwSerial);//initializing mp3 player
+ 
+  Serial.print("Ringing:");
+  Serial.println(myhours2);
   if (myhours2 > 12) {
     myhours2 -= 12;
   }
-  Serial.print("Ringing:");
-  Serial.println(myhours2);
-  mp3.volume(27);
+
+  myDFPlayer.volume(25);
   for (int n = 1; n <= myhours2; n++) {
-    if (n == 1) {
-      mp3.play(1);
-    }  //ring the chime}
-    else { mp3.play(2); }
+    if(n==1){
+    myDFPlayer.play(1);} //ring the chime}
+    else{myDFPlayer.play(2);}
     Serial.print("Ring nr:");
     Serial.println(n);
     delay(2250);
   }
+  ~myDFPlayer.begin(); //wondering if it's really necessary here, as we cut the power anyway
```

Comment:
If we are sure that we will go to sleep afterwards, then I suppose destructor is not needed.

Because every variable without `RTC_DATA_ATTR` will be cleared.

Reply:
Agreed